front/src/modules/timetableItem/components/Timetable/useFilterTimetableItems.ts

```diff
@@ -146,21 +147,33 @@ const useFilterTimetableItems = (
         if (trainCategoryFilter !== 'all') {
           const exceptionsCategories = isPacedTrainWithDetails(timetableItem)
             ? timetableItem.exceptions
-                .map(
-                  (exception) =>
-                    exception.rolling_stock_category?.value &&
-                    isMainCategory(exception.rolling_stock_category.value) &&
-                    extractTagCode(exception.rolling_stock_category.value.main_category)
-                )
-                .filter((tagCode) => !!tagCode)
+                .map((exception) => {
+                  const value = exception.rolling_stock_category?.value;
+                  if (value && isMainCategory(value)) {
+                    return extractTagCode(value.main_category);
+                  }
+                  return null;
+                })
+                .filter((tagCode): tagCode is string => !!tagCode)
```

Comment:
`tagCode is string` is not needed I think

Reply:
I get the following error if I don't put it in: `'tagCode' refers to a value, but is being used as a type here. Did you mean 'typeof tagCode'?ts(2749)`

Reply:
I don't have it when doing this 🤔 
```suggestion
                .filter((tagCode) => !!tagCode)
```